### PR TITLE
macOS: Update Menu font

### PR DESCRIPTION
**Pull request by chiuam**
### Platforms Impacted
- [ ] iOS
- [X] macOS
- [ ] win32 (Office)
- [ ] windows
- [ ] android

### Description of changes

Updating the font token value for MenuItem to align with ContextualMenu visuals.

### Verification

| Before                                       | After                                      |
|----------------------------------------------|--------------------------------------------|
| image |image |
ContextualMenu:
image

### Pull request checklist

This PR has considered (when applicable):
- [ ] Automated Tests
- [ ] Documentation and examples
- [ ] Keyboard Accessibility
- [ ] Voiceover
- [ ] Internationalization and Right-to-left Layouts

### CODE REVIEW

File: packages/components/Menu/src/MenuItem/MenuItemTokens.macos.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ export const defaultMenuItemTokens: TokenSettings<MenuItemTokens, Theme> = (t: T
   checkmarkSize: 16,
   color: t.colors.menuItemText, // matches ContextualMenu
   fontFamily: t.typography.families.primary,
-  fontSize: globalTokens.font.size300,
+  fontSize: 13, // aligning with ContextualMenu font size
```
Comment (mischreiber) [inline packages/components/Menu/src/MenuItem/MenuItemTokens.macos.ts:12]: Is it worth noting that, not only are we aligning to ContextualMenu, but also to the native macOS menu?

Comment (Saadnajmi) [inline packages/components/Menu/src/MenuItem/MenuItemTokens.macos.ts:12]: Update comment with :and/or align with NSMenu?

**Review by Saadnajmi — APPROVED** [review]

File: packages/components/Menu/src/MenuItem/MenuItemTokens.macos.ts (hunk 1 above)
Comment (Saadnajmi) [reply packages/components/Menu/src/MenuItem/MenuItemTokens.macos.ts:12]: We have the same comment :D

Comment (Saadnajmi) [reply packages/components/Menu/src/MenuItem/MenuItemTokens.macos.ts:12]: Confirmed with HIG that 13 is correct. Probably why I used it for ContextualMenu. If you wanted to be very exact, you could also check with this: https://developer.apple.com/documentation/appkit/nsfont/1533068-menufont

Comment (chiuam) [reply packages/components/Menu/src/MenuItem/MenuItemTokens.macos.ts:12]: @Saadnajmi Thank you for checking! I'm thinking replacing ContextualMenu with just NSMenu assuming that CM will soon be deprecated?

Comment (Saadnajmi) [reply packages/components/Menu/src/MenuItem/MenuItemTokens.macos.ts:12]: Yeah that sounds fine. Remove references to CM and replace with NSMenu

**Pull request merged**